### PR TITLE
make User object hashable to allow use with sets

### PR DESCRIPTION
**Pull request by mef51**
this allows creating sets out of the results of queries and enables useful set operations and one-liners like 
```connections = list(set(followers) & set(friends))```

### CODE REVIEW

File: tweepy/models.py [hunk 1]
```diff
@@ -224,7 +224,12 @@ def __ne__(self, other):
             return result
 
         return not result
-
+    
+    def __hash__(self):
+        if hasattr(self, 'id'):
+            return hash(self.id)
```
Comment (Harmon758) [inline tweepy/models.py:230]: Using `hash` is unnecessary here, since `User.id` is already an integer.
```suggestion
            return self.id
```

**Review by Harmon758 — CHANGES_REQUESTED** [review]
I think it should be safe to assume `User.id` exists, as in `__eq__`.

File: tweepy/models.py [hunk 2]
```diff
@@ -211,6 +211,12 @@ def __eq__(self, other):
 
         return NotImplemented
 
+    def __hash__(self):
+        if hasattr(self, 'id'):
+            return self.id
+        else:
+            raise TypeError('unhashable type: {} (no id attribute)'.format(type(self)))
```
Comment (Harmon758) [inline tweepy/models.py:218]:
```suggestion
        return self.id
```

**Review by Harmon758 — CHANGES_REQUESTED** [review]

**Review by Harmon758 — APPROVED** [review]

**Comment by mef51** [comment]
Thanks!

**Pull request merged**